component/src/main/java/org/wso2/extension/siddhi/io/report/util/QueryModeDataProvider.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.report.util;
+
+import ar.com.fdvs.dj.domain.builders.ColumnBuilder;
+import ar.com.fdvs.dj.domain.builders.DynamicReportBuilder;
+import ar.com.fdvs.dj.domain.entities.columns.AbstractColumn;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class is the implementation of the query mode data provider.
+ */
+public class QueryModeDataProvider implements DataProvider {
+    private static Logger logger = Logger.getLogger(QueryModeDataProvider.class);
+    private HikariDataSource dataSource;
+    private String dataSourceName;
+    private List<AbstractColumn> abstractColumns;
+    private Map<String, AbstractColumn> abstractColumnMap;
+    private DynamicReportBuilder reportBuilder;
+
+    public QueryModeDataProvider(String dataSourceName) {
+        this.dataSourceName = dataSourceName;
+        this.dataSource = RDBMSUtil.getDataSourceService(this.dataSourceName);
+    }
+
+    public void setReportBuilder(DynamicReportBuilder reportBuilder) {
+        this.reportBuilder = reportBuilder;
+    }
+
+    private Connection getConnection() {
+        Connection conn;
+        try {
+            conn = this.dataSource.getConnection();
+        } catch (SQLException e) {
+            throw new SiddhiAppRuntimeException("Cannot initialize datasource '" + this.dataSourceName +
```

Comment:
This is not datasource initialization, so lets change the exception message
`Cannot create connection from datasource + this.dataSourceName + .`

Reply:
Fixed with 80af72bd5ec9f9ff8716c1b93dc42c63205ad75a